doc/feature/cppPlugins.md

```diff
@@ -0,0 +1,159 @@
+C++ Plugin API for CPU compressors  {#data}
+============
+
+This document specifies a modern API for the in-memory C plugin API for
+lossless byte data compressors. It will not replace the 2D image
+compressors and CPU-GPU transfer plugins.
+
+## Requirements
+
+* Support all features used by current lossless byte compressors
+    * Lossless compression of binary blobs
+    * Multiple compressors per plugin DSO
+* Information provided for each compressor:
+    * Compression speed (relative to RLE compressor)
+    * Compression ratio (relative to uncompressed data)
```

Comment:
The spec should say which reference data is used for measuring the relative speed and ratio.